includes/LogisticsOrderFactory.php

```diff
@@ -9,21 +9,151 @@
      * @return LogisticsOrder
      */
-    public static function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    public function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    {
+        $logisticsOrder = new LogisticsOrder();
+        $logisticsOrder->setUrl($this->settingsRepository->getSiteUrl() . "/wp-json/gebrueder-weiss-woocommerce/v1/update/" . $wooCommerceOrder->get_id());
+        $logisticsOrder->setCreationDateTime($wooCommerceOrder->get_date_created());
+
+        $logisticsOrder->setLogisticsAddresses([
+            $this->createConsigneeAddress($wooCommerceOrder),
+            $this->createOrderByAddress()
+        ]);
+
+        $logisticsOrder->setOrderLines(
+            $this->createOrderLines($wooCommerceOrder)
+        );
+
+        return $logisticsOrder;
+    }
+
+    /**
+     * Creates the consignee address from a WooCommerce Order.
+     *
+     * @param object $wooCommerceOrder The WooCommerce order.
+     * @return LogisticsAddress
+     */
+    private function createConsigneeAddress(object $wooCommerceOrder): LogisticsAddress
+    {
+        $logisticsAddress = new LogisticsAddress();
+        $logisticsAddress->setAddressType("consignee");
+
+        $address = new Address();
+        $address->setName1($wooCommerceOrder->get_shipping_first_name());
+        $address->setName2($wooCommerceOrder->get_shipping_last_name());
+        $address->setName3($wooCommerceOrder->get_shipping_company());
+        $address->setStreet1($wooCommerceOrder->get_shipping_address_1());
+        $address->setStreet2($wooCommerceOrder->get_shipping_address_2());
+        $address->setCity($wooCommerceOrder->get_shipping_city());
+        $address->setZipCode($wooCommerceOrder->get_shipping_postcode());
+        $address->setCountryCode($wooCommerceOrder->get_shipping_country());
+        $address->setState($wooCommerceOrder->get_shipping_state());
+        $logisticsAddress->setAddress($address);
+
+        $fullName = implode(" ", array_filter([$wooCommerceOrder->get_shipping_first_name(), $wooCommerceOrder->get_shipping_last_name()]));
+
+        $contact = new Contact();
+        $contact->setName($fullName);
+        $contact->setEmail($wooCommerceOrder->get_billing_email());
+        $contact->setPhone($wooCommerceOrder->get_billing_phone());
+        $contact->setLanguage("de-De");
+        $logisticsAddress->setContact($contact);
+
+        return $logisticsAddress;
+    }
+
+    /**
+     * Creates the order by address from
+     *
+     * @return LogisticsAddress
+     */
+    private function createOrderByAddress(): LogisticsAddress
+    {
+        $logisticsAddress = new LogisticsAddress();
+        $logisticsAddress->setAddressType("orderby");
+
+        $addressReference = new AddressReference();
+        $addressReference->setQualifier("gwcustomerid");
+        $addressReference->setReference($this->settingsRepository->getCustomerId());
```

Comment:
what does this addressReference do?

Reply:
Good catch. I forgot to attach this one to the logistics address.